Client/Helpers/MsalAuthenticationProvider.cs

```diff
@@ -0,0 +1,52 @@
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Threading.Tasks;
+using Microsoft.Identity.Client;
+using Microsoft.Graph;
+
+namespace DeltaQueryClient
+{
+    // This class encapsulates the details of getting a token from MSAL and exposes it via the 
+    // IAuthenticationProvider interface so that GraphServiceClient or AuthHandler can use it.
+    // A significantly enhanced version of this class will in the future be available from
+    // the GraphSDK team.  It will supports all the types of Client Application as defined by MSAL.
+    public class MsalAuthenticationProvider : IAuthenticationProvider
+    {
+        private PublicClientApplication _clientApplication;
+        private string[] _scopes;
+        private string _accessToken = null;
+
+        public MsalAuthenticationProvider(PublicClientApplication clientApplication, string[] scopes) {
+            _clientApplication = clientApplication;
+            _scopes = scopes;
+        }
+
+        /// <summary>
+        /// Update HttpRequestMessage with credentials
+        /// </summary>
+        public async Task AuthenticateRequestAsync(HttpRequestMessage request)
+        {
+            var token = await GetTokenAsync();
+            request.Headers.Authorization = new AuthenticationHeaderValue("bearer", token);
+        }
+
+        /// <summary>
+        /// Acquire Token 
+        /// </summary>
+        public async Task<string> GetTokenAsync()
+        {
+            if (_accessToken == null)
+            {
+                AuthenticationResult authResult = null;
+                authResult = await _clientApplication.AcquireTokenAsync(_scopes);
```

Comment:
Also this can throw. we need to handle the exception;

My proposal: let's work together on a class that you could reuse in all the samples.

Reply:
Also as you suggested, I'm happy to work together on this to make this class reusable.

Reply:
@jmprieur If we build an AuthorizationProvider class that works for all the samples, at what point are we reimplementing the classes we are doing in https://github.com/microsoftgraph/msgraph-sdk-dotnet-auth ?  Perhaps we should focus on getting those completed faster, so the samples can take advantage of them?

Reply:
I agree @darrelmiller